sky/serve/core.py

```diff
@@ -156,6 +157,18 @@ def up(
             controller=controller_utils.Controllers.SKY_SERVE_CONTROLLER,
             task_resources=task.resources)
 
+        # Check that the Envoy load balancer isn't being used on an unsupported
+        # cloud.
+        lb_type = task_config.get('service', {}).get('load_balancer_type', None)
```

Comment:
could we do this as a follow up PR? Since we've already got the load_balancer_policy field I think it would make more sense.

That said if you prefer I can tack it on to this one

Reply:
I think the more PR we separate it to, the more backward compatibility we need to tackle in the future. But cc @Michaelvll for some inputs here